packages/cli/src/commands/apps/update.ts

```diff
@@ -32,22 +31,24 @@ export default class AppUpdateCommand extends InputOutputAPICommand<AppRequest,
 		const { args, argv, flags } = this.parse(AppUpdateCommand)
 		await super.setup(args, argv, flags)
 
-		this.processNormally(async data => {
-			if (flags.authorize) {
-				if (data.lambdaSmartApp) {
-					if (data.lambdaSmartApp.functions) {
-						const requests = data.lambdaSmartApp.functions.map((it) => {
-							return addPermission(it)
-						})
-						await Promise.all(requests)
+		this.processNormally(args.id,
+			() => { return this.client.apps.list() },
+			async (id, data) => {
+				if (flags.authorize) {
+					if (data.lambdaSmartApp) {
+						if (data.lambdaSmartApp.functions) {
+							const requests = data.lambdaSmartApp.functions.map((it) => {
+								return addPermission(it)
+							})
+							await Promise.all(requests)
+						}
+					} else {
+						this.logger.error('Authorization is not applicable to web-hook SmartApps')
+						// eslint-disable-next-line no-process-exit
+						process.exit(1)
```

Comment:
You should just be able to `throw Error('Authorization...` instead of logging and exiting and `processNormally` will log the error and exit for you.